File: sdk/src/main/java/com/sygic/travel/sdk/model/api/PlacesResponse.java
package com.sygic.travel.sdk.model.api;

import com.google.gson.JsonObject;
import com.google.gson.reflect.TypeToken;
import com.sygic.travel.sdk.Parser;
import com.sygic.travel.sdk.model.place.Place;

import java.util.List;

/**
 * <p>Response that contains a list of basic places data (without detailed data). Suitable for showing
 * places in a list or spread on a map.</p>
 */
public class PlacesResponse extends StResponse {
	private static final String PLACES = "places";
	private List<Place> places;

	private JsonObject data;

	public Object getData() {
		if(places == null && data != null) {
			places = Parser.parseJson(
				data.get(PLACES).toString(),
				new TypeToken<List<Place>>(){}.getType()
			);
		}
		return places;
	}
}

File: sdk/src/main/java/com/sygic/travel/sdk/api/responseWrappers/PlacesResponse.kt
package com.sygic.travel.sdk.api.responseWrappers

import com.google.gson.annotations.SerializedName
import com.sygic.travel.sdk.api.model.place.ApiPlaceDetail
import com.sygic.travel.sdk.model.place.Place

/**
 * Response that contains a list of places data. Suitable for showing places in a list or
 * spread on a map.
 */
internal class PlacesResponse : StResponse() {
	private var data: Data? = null

	fun getPlaces(): List<Place>? {
		val places = data?.apiPlaces
		val convertedPlaces: MutableList<Place> = mutableListOf()

		places?.mapTo(convertedPlaces) {
			it.convert()
		}

		return convertedPlaces
	}

	inner class Data {
		@SerializedName("places")
		var apiPlaces: List<ApiPlaceDetail>? = null
	}
}
